### PR TITLE
add sha to report and code to check it

### PR DESCRIPTION
**Pull request by mmulholla**
Calculates a sha value for a report and adds it to the report in the tool metadata.

```
apiversion: v1
kind: verify-report
metadata:
    tool:
        verifier-version: 1.9.0
        report-digest: uint64:3746509530969125134
```  

when the chart-verifier report option is used to generate a summary of a given report the ```report-digest``` value is compared to an equivalent digest value calculated for the report to make sure they match. If they do not match an error is output:
```Digest in report does not match report content, report has been modified.```

To support this a change was also made to include a GetVersion api and the release file was moved:

from: ```cmd/release/release_info.json```
to:  ```pkg/chartverifier/version/version_info.json```

documentation and scripts were modified to reflect this change.

To prevent failures with older reports the digest check is only performed if the report version is 1.9.0 or greater (current version is 1.8.0 and this digest feature is expected to GA in 1.9.0) 

Note: Because of the 1.9.0 check this change should not cause regressions in existing tests, until after 1.9.0 is released.

### CODE REVIEW

File: docs/helm-chart-checks.md [hunk 1]
```diff
@@ -32,6 +32,7 @@ Helm chart checks are a set of checks against which the Red Hat Helm chart-verif
   - Alternatively the ```--write-to-file``` flag can be used to write to a ```report.yaml``` file. 
 - An error log is created for all verify commands but can be optionally suppressed.
 - You can indicate that a chart is not to be published in the OpenShift catalog.
+- From chart verifier version 1.9.0 the generated report includes a sha value based on the report content. This is used during the submission process to verify the egrity of the report.
```
Comment (isutton) [inline docs/helm-chart-checks.md:35]:
```suggestion
- From chart verifier version 1.9.0 the generated report includes a sha value based on the report content. This is used during the submission process to verify the integrity of the report.
```

File: pkg/chartverifier/report/types.go [hunk 2]
```diff
@@ -26,6 +26,7 @@ type ReportMetadata struct {
 type ToolMetadata struct {
 	Version                    string  `json:"verifier-version" yaml:"verifier-version"`
 	Profile                    Profile `json:"profile" yaml:"profile"`
+	ReportDigest               string  `json:"report-digest" yaml:"report-digest"`
```
Comment (isutton) [inline pkg/chartverifier/report/types.go:29]: Even though other keys are using `kebab-case` names, `snakeCase` should be used instead.

File: pkg/chartverifier/reportsummary/test-reports/report.json [hunk 3]
```diff
@@ -1 +1 @@
...
...ndatory","outcome":"PASS","reason":"Image is Red Hat certified : registry.access.redhat.com/rhscl/postgresql-10-rhel7:1-66"},{"check":"v1.0/chart-testing","type":"Mandatory","outcome":"PASS","reason":"Chart tests have passed"},{"check":"v1.0/has-readme","type":"Mandatory","outcome":"PASS","reason":"Chart has a README"},{"check":"v1.0/is-helm-v3","type":"Mandatory","outcome":"PASS","reason":"API version is V2, used in Helm 3"},{"check":"v1.0/contains-test","type":"Mandatory","outcome":"PASS","reason":"Chart test files exist"},{"check":"v1.0/contains-values","type":"Mandatory","outcome":"PASS","reason":"Values file exist"},{"check":"v1.0/required-annotations-present","type":"Mandatory","outcome":"PASS","reason":"All required annotations present"},{"check":"v1.0/contains-values-schema","type":"Mandatory","outcome":"PASS","reason":"Values schema file exist"},{"check":"v1.1/has-kubeversion","type":"Mandatory","outcome":"PASS","reason":"Kubernetes version specified"},{"check":"v1.0/not-contains-crds","type":"Mandatory","outcome":"PASS","reason":"Chart does not contain CRDs"},{"check":"v1.0/not-contain-csi-objects","type":"Mandatory","outcome":"PASS","reason":"CSI objects do not exist"}]}
```
Comment (tisutisu) [inline pkg/chartverifier/reportsummary/test-reports/report.json:1]: Same for this file as well.

File: docs/helm-chart-checks.md (hunk 1 above)
Comment (tisutisu) [inline docs/helm-chart-checks.md:35]: `egrity` a type here.

Comment (mmulholla) [reply docs/helm-chart-checks.md]: Thx.

File: pkg/chartverifier/reportsummary/test-reports/report.json (hunk 3 above)
Comment (mmulholla) [reply pkg/chartverifier/reportsummary/test-reports/report.json]: Done

File: pkg/chartverifier/reportsummary/test-reports/predigest/report.json [hunk 4]
```diff
@@ -0,0 +1 @@
...ndatory","outcome":"PASS","reason":"Image is Red Hat certified : registry.access.redhat.com/rhscl/postgresql-10-rhel7:1-66"},{"check":"v1.0/chart-testing","type":"Mandatory","outcome":"PASS","reason":"Chart tests have passed"},{"check":"v1.0/has-readme","type":"Mandatory","outcome":"PASS","reason":"Chart has a README"},{"check":"v1.0/is-helm-v3","type":"Mandatory","outcome":"PASS","reason":"API version is V2, used in Helm 3"},{"check":"v1.0/contains-test","type":"Mandatory","outcome":"PASS","reason":"Chart test files exist"},{"check":"v1.0/contains-values","type":"Mandatory","outcome":"PASS","reason":"Values file exist"},{"check":"v1.0/required-annotations-present","type":"Mandatory","outcome":"PASS","reason":"All required annotations present"},{"check":"v1.0/contains-values-schema","type":"Mandatory","outcome":"PASS","reason":"Values schema file exist"},{"check":"v1.1/has-kubeversion","type":"Mandatory","outcome":"PASS","reason":"Kubernetes version specified"},{"check":"v1.0/not-contains-crds","type":"Mandatory","outcome":"PASS","reason":"Chart does not contain CRDs"},{"check":"v1.0/not-contain-csi-objects","type":"Mandatory","outcome":"PASS","reason":"CSI objects do not exist"}]}
```
[the remark this answers is not in the document]
Comment (mmulholla) [reply pkg/chartverifier/reportsummary/test-reports/predigest/report.json]: Done.

File: pkg/chartverifier/report/report.go [hunk 5]
```diff
@@ -142,3 +147,42 @@ func loadReportFromRemote(url *url.URL) (string, error) {
 
 	return string(reportBytes), nil
 }
+
+func (r *Report) GetReportDigest() (string, error) {
+
+	savedDigest := r.Metadata.ToolMetadata.ReportDigest
+	r.Metadata.ToolMetadata.ReportDigest = ""
```
[the remark this answers is not in the document]
Comment (mmulholla) [reply pkg/chartverifier/report/report.go:154]: the digest is based on a report with the digest set to blank, so need to blank it put before calculating its digest. Without this adding the digest would change the digest.

File: pkg/chartverifier/report/types.go (hunk 2 above)
Comment (mmulholla) [reply pkg/chartverifier/report/types.go]: Updated.

File: docs/helm-chart-checks.md (hunk 1 above)
Comment (mmulholla) [reply docs/helm-chart-checks.md]: Updated. thx

**Comment by mmulholla** [comment]
Have added updates for code review comments but also made additional changes.

- Added an option to the report command and api to skip the digest check when generating a report summary but making it clear that is is intended for testing purposes only (David's request). Obviously the chart workflow will not set the option to skip the option will be useful in some of our tests. 
- Moved the code which performs the digest check to the report summary code to support the new option.

**Review by tisutisu — COMMENTED** [review]
LGTM

**Review by tisutisu — APPROVED** [review]
All the review comments are addressed now.

**Pull request merged**